beacon_node/http_api/src/publish_blocks.rs

```diff
@@ -205,16 +206,59 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlock<T>>(
 pub async fn publish_blinded_block<T: BeaconChainTypes>(
     block: SignedBeaconBlock<T::EthSpec, BlindedPayload<T::EthSpec>>,
     chain: Arc<BeaconChain<T>>,
+    task_spawner: TaskSpawner<T::EthSpec>,
     network_tx: &UnboundedSender<NetworkMessage<T::EthSpec>>,
     log: Logger,
     validation_level: BroadcastValidation,
 ) -> Result<(), Rejection> {
     let block_root = block.canonical_root();
-    let full_block: ProvenancedBlock<T, Arc<SignedBeaconBlock<T::EthSpec>>> =
-        reconstruct_block(chain.clone(), block_root, block, log.clone()).await?;
+
+    // Complete gossip verification and publication to the relay in parallel.
+    // This avoids tripping the gossip duplicate filter.
+    let block_to_verify = Arc::new(block.clone());
```

Comment:
This clone of the blinded block is a bit unfortunate but I think unavoidable as we need an `Arc` block for most functions, but `reconstruct_payload` needs a full block. Because the block is blinded, cloning it should be quite fast!